test/e2e/storage/testsuites/volume_modify.go

```diff
@@ -0,0 +1,310 @@
+		framework.ExpectNoError(err, "While creating pod for modifying")
+
+		ginkgo.By("Modifying PVC via VAC")
+		newPVC, err := SetPVCVACName(ctx, l.resource.Pvc, newVAC.Name, f.ClientSet, setVACWaitPeriod)
+		framework.ExpectNoError(err, "While updating VolumeAttributesClassName on PVC")
+		l.resource.Pvc = newPVC
+		gomega.Expect(l.resource.Pvc).NotTo(gomega.BeNil())
+
+		ginkgo.By("Waiting for modification to finish")
+		err = WaitForVolumeModification(ctx, l.resource.Pvc, f.ClientSet, modifyVolumeWaitPeriod)
+		framework.ExpectNoError(err, "While waiting for volume modification to finish")
+
+		pvcConditions := l.resource.Pvc.Status.Conditions
+		gomega.Expect(pvcConditions).To(gomega.BeEmpty(), "PVC should not have conditions")
+	})
+}
+
+// SetPVCVACName sets the VolumeAttributesClassName on a PVC object
+func SetPVCVACName(ctx context.Context, origPVC *v1.PersistentVolumeClaim, name string, c clientset.Interface, timeout time.Duration) (*v1.PersistentVolumeClaim, error) {
+	pvcName := origPVC.Name
+	updatedPVC := origPVC.DeepCopy()
+
+	// Retry the update on error, until we hit a timeout.
+	var lastUpdateError error
+	waitErr := wait.PollUntilContextTimeout(ctx, modifyPollInterval, timeout, true, func(ctx context.Context) (bool, error) {
+		var err error
+		updatedPVC, err = c.CoreV1().PersistentVolumeClaims(origPVC.Namespace).Get(ctx, pvcName, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error fetching pvc %q for updating: %w", pvcName, err)
+		}
+
+		updatedPVC.Spec.VolumeAttributesClassName = &name
+		updatedPVC, err = c.CoreV1().PersistentVolumeClaims(origPVC.Namespace).Update(ctx, updatedPVC, metav1.UpdateOptions{})
+		if err != nil {
+			framework.Logf("Error updating pvc %s: %v", pvcName, err)
+			lastUpdateError = err
+			return false, nil
+		}
+		return true, nil
+	})
+	if wait.Interrupted(waitErr) {
+		return nil, fmt.Errorf("timed out attempting to update PVC VAC name. last update error: %w", lastUpdateError)
+	}
+	if waitErr != nil {
+		return nil, fmt.Errorf("failed to modify PVC (check logs for error): %w", waitErr)
+	}
+	return updatedPVC, nil
+}
+
+// WaitForVolumeModification waits for the volume to be modified
+// The input PVC is assumed to have a VolumeAttributesClassName set
+func WaitForVolumeModification(ctx context.Context, pvc *v1.PersistentVolumeClaim, c clientset.Interface, timeout time.Duration) error {
+	pvName := pvc.Spec.VolumeName
+	waitErr := wait.PollUntilContextTimeout(ctx, modifyPollInterval, timeout, true, func(ctx context.Context) (bool, error) {
+		pv, err := c.CoreV1().PersistentVolumes().Get(ctx, pvName, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error fetching pv %q: %w", pvName, err)
+		}
+
+		// If PVC VAC name and PV VAC name match, modification is done
+		if pv.Spec.VolumeAttributesClassName != nil && *pv.Spec.VolumeAttributesClassName == *pvc.Spec.VolumeAttributesClassName {
+			return true, nil
+		}
+		return false, nil
```

Comment:
please also check whether pvc.status.volumeModifyStatus is nil and pvc.status.currentVAC is equal to pvc.spec.vac.

Reply:
Fixed in next revision (also refacted this check out to a function called `vacMatches` so it can be used elsewhere in this file)